program-test/src/lib.rs

```diff
@@ -698,3 +700,62 @@ impl ProgramTestBanksClientExt for BanksClient {
         ))
     }
 }
+
+struct DroppableTask<T>(Sender<()>, JoinHandle<T>);
+
+impl<T> Drop for DroppableTask<T> {
+    fn drop(&mut self) {
+        match self.0.try_send(()) {
+            Ok(_) => {}
+            Err(e) => {
+                debug!("Error cancelling task: {:?}", e);
+            }
+        }
+    }
+}
+
+pub struct ProgramTestState {
+    pub banks_client: BanksClient,
+    pub payer: Keypair,
+    pub last_blockhash: Hash,
+    _bank_task: DroppableTask<()>,
+}
+
+impl ProgramTestState {
+    pub fn new(
```

Comment:
I think we can limit scope here
```suggestion
    fn new(
```

Reply:
Good call, I was thinking someone might want to hack on it from the outside, but this'll be safer